Sources/Queues/Job.swift

```diff
@@ -6,105 +6,130 @@ import Vapor
 
-    /// Called when there was an error and job will be retired.
+    /// Called when there was an error and the job will be retried.
     ///
-    /// - Parameters:
-    ///     - attempt: Number of job attempts which failed
-    /// - Returns: Number of seconds for which next retry will be delayed.
-    ///   Return `-1` if you want to retry job immediately without putting it back to the queue.
+    /// - Parameter attempt: Number of job attempts which have failed so far.
+    /// - Returns: Number of seconds to delay the next retry. Return `-1` to retry immediately without returning the
+    ///   job to the queue first. Return `0` to place the job back on the queue with no delay added. If this method
+    ///   is not implemented, the default is `0`.
     func nextRetryIn(attempt: Int) -> Int
     
+    /// Serialize a typed payload to an array of bytes. When `Payload` is `Codable`, this method will default to
+    /// encoding to JSON.
+    ///
+    /// - Parameter payload: The payload to serialize.
+    /// - Returns: The array of serialized bytes.
     static func serializePayload(_ payload: Payload) throws -> [UInt8]
+    
+    /// Deserialize an array of bytes into a typed payload. When `Payload` is `Codable`, this method will default to
+    /// decoding from JSON.
+    ///
+    /// - Parameter bytes: The serialized bytes to decode.
+    /// - Returns: A decoded payload.
     static func parsePayload(_ bytes: [UInt8]) throws -> Payload
 }
 
-extension Job where Payload: Codable {
-    
-    /// Serialize a payload into Data
-    /// - Parameter payload: The payload
+extension Job where Payload: Codable {    
+    /// Default implementation for ``Job/serializePayload(_:)-4uro2``.
     public static func serializePayload(_ payload: Payload) throws -> [UInt8] {
         try .init(JSONEncoder().encode(payload))
     }
     
-    /// Parse bytes into the payload
-    /// - Parameter bytes: The Payload
+    /// Default implementation for ``Job/parsePayload(_:)-9tn3a``.
     public static func parsePayload(_ bytes: [UInt8]) throws -> Payload {
         try JSONDecoder().decode(Payload.self, from: .init(bytes))
     }
 }
 
 extension Job {
-    /// The jobName of the Job
+    /// Default implementation for ``AnyJob/name``.
     public static var name: String {
-        return String(describing: Self.self)
+        String(describing: Self.self)
     }
     
-    /// See `Job`.`error`
+    /// Default implementation for ``Job/error(_:_:_:)-jzgw``.
     public func error(
         _ context: QueueContext,
-        _ error: Error,
+        _ error: any Error,
         _ payload: Payload
     ) -> EventLoopFuture<Void> {
-        context.eventLoop.makeSucceededFuture(())
+        context.eventLoop.makeSucceededVoidFuture()
     }
 
-    /// See `Job`.`nextRetryIn`
+    /// Default implementation for ``Job/nextRetryIn(attempt:)-5gc93``.
     public func nextRetryIn(attempt: Int) -> Int {
-        return -1
+        0
     }
```

Comment:
are you sure you want to change that from `-1` to `0`? I'm worried it'll have some unintended effects.

Reply:
It's a deliberate choice. `-1` no longer has any special meaning (and I should probably update the doc comments to reflect that)